unittests/baseplatformtests.py

```diff
@@ -311,28 +315,31 @@ def get_compdb(self):
                     each['command'] = compiler + ' ' + f.read()
         return contents
 
+    def get_meson_log_raw(self):
+        return self._get_meson_log()
```

Comment:
That would mean extra unnecessary code on the call sites though for assembling the contents of the log into a string, so it seems easier to just use `get_meson_log_raw` for them, no? Even though it has a potential unhandled error case, so did the previous code.

Reply:
The previous code would raise a FileNotFoundError, right? And the new code would instead raise
```
TypeError: expected string or bytes-like object
```

But I don't see what unnecessary code you are referring to here. It would still return the contents of the log as a string, using the function is fine. Just like we have:
```python
     def get_meson_log(self):
        with self._open_meson_log() as f:
            return f.readlines()
```

so too:
```python
     def get_meson_log_raw(self):
        with self._open_meson_log() as f:
            return f.read()
```

Reply:
Oh I thought you meant the callers should use (a public variant of) _open_meson_log. Indeed this makes sense to get nicer Exceptions, will adjust the code.

Reply:
But also `_get_meson_log()` is a silly name and we should call it `maybe_get_meson_log()`. :D

Reply:
@eli-schwartz I think this is solved, if I understood your remark correctly.